transformer_engine/pytorch/attention.py

```diff
@@ -1286,12 +1287,24 @@ def forward(
         cu_seqlens_q: Optional[torch.Tensor] = None,
         cu_seqlens_kv: Optional[torch.Tensor] = None,
         attn_mask_type: str = "causal",
+        window_size: Optional[Tuple[int, int]] = None,
         cp_group: Optional[dist_group_type] = None,
         cp_global_ranks: List[int] = None,
         cp_stream: torch.cuda.Stream = None,
     ) -> torch.Tensor:
         """flash-attn fprop"""
 
+        if "causal" in attn_mask_type:
+            if window_size is None:
```

Comment:
This piece of code is replicated in multiple places, can we refactor that into a function?

Reply:
Done.